witness/golang/omniwitness/internal/omniwitness/omniwitness.go

```diff
@@ -148,13 +182,93 @@ func Main(ctx context.Context, signer note.Signer, httpListener net.Listener, ht
 		Handler: r,
 	}
 	g.Go(func() error {
-		defer srv.Shutdown(ctx)
+		glog.Info("HTTP server goroutine started")
+		defer glog.Info("HTTP server goroutine done")
 		return srv.Serve(httpListener)
 	})
+	g.Go(func() error {
+		// This goroutine brings down the HTTP server when ctx is done.
+		glog.Info("HTTP server-shutdown goroutine started")
+		defer glog.Info("HTTP server-shutdown goroutine done")
+		<-ctx.Done()
+		return srv.Shutdown(ctx)
+	})
 
 	return g.Wait()
 }
 
+func runDistributors(ctx context.Context, g *errgroup.Group, logs []dist_gh.Log, witness dist_gh.Witness, operatorConfig OperatorConfig) {
+	distribute := func(opts *dist_gh.DistributeOptions) error {
+		if err := dist_gh.DistributeOnce(ctx, opts); err != nil {
+			glog.Errorf("DistributeOnce: %v", err)
+		}
+		for {
+			select {
+			case <-time.After(distributeInterval):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+			if err := dist_gh.DistributeOnce(ctx, opts); err != nil {
+				glog.Errorf("DistributeOnce: %v", err)
+			}
+		}
+	}
+
+	g.Go(func() error {
+		dr, err := github.NewRepoID("mhutchinson/mhutchinson-distributor")
+		if err != nil {
+			return fmt.Errorf("NewRepoID: %v", err)
+		}
+		glog.Infof("Distributor %q goroutine started", dr)
+		defer glog.Infof("Distributor %q goroutine done", dr)
+
+		fork, err := github.NewRepoID(fmt.Sprintf("%s/mhutchinson-distributor", operatorConfig.GithubUser))
+		if err != nil {
+			return fmt.Errorf("NewRepoID: %v", err)
+		}
+		repo, err := github.NewRepository(ctx, dr, "main", fork, operatorConfig.GithubUser, operatorConfig.GithubEmail, operatorConfig.GithubToken)
+		if err != nil {
+			return fmt.Errorf("NewRepository: %v", err)
+		}
+		opts := &dist_gh.DistributeOptions{
+			Repo:            repo,
+			DistributorPath: "distributor",
+			Logs:            logs,
+			WitSigV:         operatorConfig.WitnessVerifier,
+			Witness:         witness,
+		}
+
+		return distribute(opts)
+	})
+
+	g.Go(func() error {
+		dr, err := github.NewRepoID("WolseyBankWitness/rediffusion")
+		if err != nil {
+			return fmt.Errorf("NewRepoID: %v", err)
+		}
+		glog.Infof("Distributor %q goroutine started", dr)
+		defer glog.Infof("Distributor %q goroutine done", dr)
+
+		fork, err := github.NewRepoID(fmt.Sprintf("%s/rediffusion", operatorConfig.GithubUser))
+		if err != nil {
+			return fmt.Errorf("NewRepoID: %v", err)
+		}
+		repo, err := github.NewRepository(ctx, dr, "main", fork, operatorConfig.GithubUser, operatorConfig.GithubEmail, operatorConfig.GithubToken)
+		if err != nil {
+			return fmt.Errorf("NewRepository: %v", err)
+		}
+		opts := &dist_gh.DistributeOptions{
+			Repo:            repo,
+			DistributorPath: ".",
+			Logs:            logs,
+			WitSigV:         operatorConfig.WitnessVerifier,
+			Witness:         witness,
+		}
+
+		return distribute(opts)
+	})
+}
```

Comment:
Could this be factored out and shared between the two distributor goroutines?

Reply:
When writing this initially I created a top-level function to do this and it just felt like a giant handbag that took as many arguments as it had lines of code within it. I've extracted another function in this function to create the configuration, which I think is a reasonable balance. Having it as a nested function allows it to capture all of the args to `runDistributors` that it also needs, so saves redeclaring all of that.